sdks/java/core/src/main/java/org/apache/beam/sdk/runners/inprocess/InProcessEvaluationContext.java

```diff
@@ -191,10 +197,10 @@ private InProcessEvaluationContext(
       // Empty bundles don't impact watermarks and shouldn't trigger downstream execution, so
       // filter them out
       if (!Iterables.isEmpty(committed.getElements())) {
-        completed.add(committed);
+        outputs.put(committed, valueToConsumers.get(committed.getPCollection()));
```

Comment:
This changes in the immediate follow-up to this CL, which allows a TransformEvaluator to return some elements as "unprocessed". Those elements are added to the result map here, but should only be consumed by the producing transform

Ex: [`7da8e1a`](https://github.com/tgroh/incubator-beam/blob/7da8e1ad510e02d3b317f40e89310824fd99429c/sdks/java/core/src/main/java/org/apache/beam/sdk/runners/inprocess/InProcessEvaluationContext.java#L192)